### ISSUE TITLE
Server Crash when cancelling request on streamText with (Bun)

### ISSUE THREAD

**Issue by GautierT** [vercel/ai#3278]
### Description

Minimal repro in this repo : 
https://github.com/GautierT/elysia-server-crash-on-stream-stop

### Code example

Repro here : https://github.com/GautierT/elysia-server-crash-on-stream-stop

## With Bun (Crash)

Run 

`bun run start-bun`

Call POST /

```
curl --location --request POST 'http://localhost:3000'
```

then cancel the request. (ctrl+c)

Server will crash.

## With Node (Okay)

With node it's working

Run 

`bun run start-node`

Call POST /

```
curl --location --request POST 'http://localhost:3000'
```

then cancel the request. (ctrl+c)

Server is okay.

**Comment by lgrammel** [comment]
This is most likely a bun issue. Is there any error message?

**Comment by lgrammel** [comment]
Most likely this is due to Bun not fully implementing the standard JS functions. E.g. afaik there were gaps around https://developer.mozilla.org/en-US/docs/Web/API/TransformStream

**Comment by lgrammel** [comment]
https://github.com/oven-sh/bun/issues?q=is%3Aissue%20state%3Aopen%20stream is quite rich, so I will close this from the AI SDK side.

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **None**  
Confidence: **High**

### Rationale
This was an external Bun runtime defect rather than broken AI SDK behavior, so it does not represent a bug or requested change in vercel/ai.

### Evidence
- The crash occurred only under Bun when cancelling a streamed request; Node continued normally.
- The maintainer attributed the failure to Bun's incomplete web-standard stream implementation and stated there was nothing to fix in the AI SDK.
- The reporter confirmed the issue was fixed by Bun v1.1.34 ReadableStream specification updates.
